packages/clients/src/sui/gas.ts

```diff
@@ -131,27 +157,68 @@ export class GasStationClient extends ShinamiRpcClient {
 
 /**
  * Builds a gasless transaction.
- * @param sui Sui JSON RPC provider.
- * @param txb Optional base `TransactionBlock`. An empty one will be used if not specified.
- * @param build Optional builder function to further populate the base `TransactionBlock`.
- * @returns Base64 encoded gasless transaction bytes that can be passed to `sponsorTransactionBlock`
- *    request.
+ * @param tx `Transaction` object pre-populated with the target transaction data.
+ * @param options Options
+ *    - sender - Optional sender address. Can also be set on `tx`.
+ *    - gasBudget - Optional gas budget. Can also be set on `tx`.
+ *    - gasPrice - Optional gas price. Can also be set on `tx`.
+ *    - sui - `SuiClient`. Required if `tx` is not fully resolved.
+ * @returns A gasless transaction to be sponsored.
  */
-export async function buildGaslessTransactionBytes({
-  sui,
-  txb,
-  build,
-}: {
-  sui: SuiClient;
-  txb?: TransactionBlock;
-  build?: (txb: TransactionBlock) => Promise<void>;
-}): Promise<string> {
-  const _txb = txb ?? new TransactionBlock();
-  if (build) await build(_txb);
-  return toB64(
-    await _txb.build({
-      client: sui,
-      onlyTransactionKind: true,
-    }),
-  );
+export async function buildGaslessTransaction(
+  tx: Transaction,
+  options?: {
+    sender?: string;
+    gasBudget?: number | string;
+    gasPrice?: number | string;
+    sui?: SuiClient;
+  },
+): Promise<GaslessTransaction>;
+
+/**
+ * Builds a gasless transaction.
+ * @param build Builder function to populate transaction data.
+ * @param options Options
+ *    - sender - Optional sender address. Can also be set during `build`.
+ *    - gasBudget - Optional gas budget. Can also be set during `build`.
+ *    - gasPrice - Optional gas price. Can also be set during `build`.
+ *    - sui - `SuiClient`. Required if `build` uses non fully resolved inputs.
+ * @returns A gasless transaction to be sponsored.
+ */
+export async function buildGaslessTransaction(
+  build: (tx: Transaction) => void | Promise<void>,
+  options?: {
+    sender?: string;
+    gasBudget?: number | string;
+    gasPrice?: number | string;
+    sui?: SuiClient;
+  },
+): Promise<GaslessTransaction>;
+
+export async function buildGaslessTransaction(
+  txOrBuild: Transaction | ((tx: Transaction) => void | Promise<void>),
+  options?: {
+    sender?: string;
+    gasBudget?: number | string;
+    gasPrice?: number | string;
+    sui?: SuiClient;
+  },
+): Promise<GaslessTransaction> {
+  let tx: Transaction;
+  if (txOrBuild instanceof Transaction) {
+    tx = txOrBuild;
+  } else {
+    tx = new Transaction();
+    await txOrBuild(tx);
```

Comment:
@jasonxh I'm getting an error using this override of the `buildGaslessTransaction` function

.../shinami-typescript-sdk/packages/clients/dist/esm/sui/gas.js:124
            yield txOrBuild(tx);
                  ^
TypeError: txOrBuild is not a function

I believe the instanceof check here might not be working correctly.

Reply:
I suspect issue with your project setup. Let's troubleshoot offline.

Reply:
Fixed offline. There was an issue with my local setup.